presto-main/src/main/java/com/facebook/presto/operator/aggregation/DecimalSumAggregation.java

```diff
@@ -160,8 +161,8 @@ public static void outputLongDecimal(LongDecimalType type, BigIntegerState state
         }
         else {
             BigDecimal sum = new BigDecimal(state.getBigInteger(), type.getScale());
-            checkOverflow(sum);
-            type.writeBigDecimal(out, sum);
+            checkOverflow(state.getBigInteger());
```

Comment:
It's basically what was done here before.
IDK why the "sum" is called "sum" though - will rename that to "value" to be less confusing ;-)